versions/3.0.md

```diff
@@ -3012,6 +3012,125 @@ schemas:
+
+In OAS 3.0, a response payload may be described to be exactly one of any number of types:
+
+```
+MyResponseType:
+  oneOf:
+  - $ref: '#/components/schemas/Cat'
+  - $ref: '#/components/schemas/Dog'
+  - $ref: '#/components/schemas/Lizard'
+```
+
+which means the paylod _must_, by validation, match exactly one of the schemas described by `Cat`, `Dog`, or `Lizard`.  In this case, a discriminator will act as a "hint" to bypass validation and selection of the matching schema which may be a costly operation, depending on the complexity of the schema. We can then describe exactly which field tells us which schema to use:
+
+
+```
+MyResponseType:
+  oneOf:
+  - $ref: '#/components/schemas/Cat'
+  - $ref: '#/components/schemas/Dog'
+  - $ref: '#/components/schemas/Lizard'
+  discriminator:
+    property: pet_type
+```
+
+The expectation now is that a property with name `pet_type` _must_ be present in the response payload, and the value will correspond to the name of a schema defined in the OAS document.  Thus the response payload:
+
+```
+{
+  "id": 12345,
+  "pet_type": "Cat"
+}
+```
+
+Will indicate that the `Cat` schema be used in conjunction with this payload.
+
+In scenarios where the value of the discriminator field does not match the schema name, an optional `mapping` definition may be used:
+
+```
+MyResponseType:
+  oneOf:
+  - $ref: '#/components/schemas/Cat'
+  - $ref: '#/components/schemas/Dog'
+  - $ref: '#/components/schemas/Lizard'
+  discriminator:
+    property: pet_type
+    mapping:
+      cat: '#/components/schemas/Cat'
+```
+
+When used in conjunction with the `anyOf` construct, the use of the discriminator can avoid ambiguity where multiple schemas may satisfy a single payload.
+
+In both the `oneOf` and `anyOf` use cases, all possible schemas must be listed explicitly.  To avoid redundancy, the discriminator may be added to a parent schema definition, and all schemas composing the parent schema in an `allOf` construct may be used as an alternate schema.
+
+For example:
+
+```
+components:
+  schemas:
+    Pet:
+      type: object
+      required:
+      - pet_type
+      properties:
+        pet_type:
+          type: string
+      discriminator:
+        propertyName: pet_type
+        mapping:
+          cachorro: Dog
+    Cat:
+      allOf:
+      - $ref: '#/components/schemas/Pet'
+      - type: object
+        # all other properties specific to a `Cat`
+        properties:
+          name:
+            type: string
+    Dog:
+      allOf:
+      - $ref: '#/components/schemas/Pet'
+      - type: object
+        # all other properties specific to a `Dog`
+        properties:
+          bark:
+            type: string
+```
+
+a payload like this:
+
+```
+{
+  "pet_type": "Cat",
+  "name": "misty"
+}
+```
+
+will indicate that the `Cat` schema be used.  Likewise this schema:
+
+```
+{
+  "pet_type": "cachorro",
+  "bark": "soft"
+}
+```
+
+will map to `Dog` because of the definition in the `mappings` element.
+
+
```

Comment:
I suggest also adding a note:

Some OpenAPI definitions use external, reusable schema references such as
```
      properties:
        pet_type:
           type: string
           enum:
             - cat
             - cachorro
             - lizard
      oneOf:
      - $ref: 'http://www.example.com/schemas/Cat/schema.json'
      - $ref: 'http://www.example.com/schemas/Doc/schema.json'
      - $ref: 'http://www.example.com/schemas/Lizard/schema.json'
      discriminator:
        propertyName: pet_type
        mapping:
          cat: 'http://www.example.com/schemas/Cat/schema.json'
          cachorro: 'http://www.example.com/schemas/Doc/schema.json'
          lizard: 'http://www.example.com/schemas/Lizard/schema.json'
```
and an implicit name mapping based on schema names is not possible.
For such cases, a `mapping` is **required**.

The above `$ref` values are JSON Schema files, not OpenAPI documents. Schema may
also be referenced from schema definitions from an external OpenAPI definition:
```
      properties:
        pet_type:
           type: string
           enum:
             - cat
             - cachorro
             - lizard
      oneOf:
      - $ref: 'http://www.example.com/library.json#/components/schema/Cat.json'
      - $ref: 'http://www.example.com/library.json#/components/schema/Doc.json'
      - $ref: 'http://www.example.com/library.json#/components/schema/Lizard.json'
      discriminator:
        propertyName: pet_type
        mapping:
          cat: $ref: 'http://www.example.com/library.json#/components/schema/Cat.json'
          cachorro: $ref: 'http://www.example.com/library.json#/components/schema/Doc.json'
          lizard: $ref: 'http://www.example.com/library.json#/components/schema/Lizard.json'
```

[in particular, citing external JSON Schema,  and not just `#/definitions/SchemaName` from other OpenAPI 2.0 documents, worked in 2.0 and we need to ensure this still works in 2.0, whether this involves discriminators or not.]